source/OptimaJet.DWKit.StarterApplication/EventDispatcher/EnitityEventHandler/BaseHookNotifier.cs

```diff
@@ -11,70 +21,186 @@
+            IResourceGraph resourceGraph,
             IDbContextResolver dbContextResolver,
-            IHubContext<ScriptoriaHub> hubContext
+            IHubContext<ScriptoriaHub> hubContext,
+            IHubContext<JSONAPIHub> dataHub
             )
         {
-            DbContext = dbContextResolver.GetContext();
-            HubContext = hubContext;
-        }
+            this.DbContext = dbContextResolver.GetContext();
+            this.HubContext = hubContext;
+            this.DataHub = dataHub;
 
-        public string GetTableName()
-        {
-            return DbContext.GetTableName(typeof(TEntity));
+            this._operationsProcessor = operationsProcessor;
+            this._serializer = serializer;
+            this._resourceGraph = resourceGraph;
+
+            var jsonApiContext = new JsonApiContext(
+                resourceGraph: this._resourceGraph,
+                httpContextAccessor: null,
+                options: new JsonApiOptions(),
+                metaBuilder: new MetaBuilder(),
+                genericProcessorFactory: null,
+                queryParser: null,
+                controllerContext: null
+            )
+            {
+                PageManager = new PageManager
+                {
+                    DefaultPageSize = 20,
+                    CurrentPage = 0,
+                    PageSize = 20
+                }
+            };
+            this._documentBuilder = new DocumentBuilder(jsonApiContext, null);
         }
 
-        public virtual IEnumerable<string> GetUpdateGroups(string tableName, string id)
+        public string GetTableName()
         {
-            return new string[] { $"/{tableName}/{id}" };
+            return this.DbContext.GetTableName(typeof(TEntity));
         }
 
         public virtual void DidDelete(string id)
         {
-            NotifyOperation(id, Delete);
+            this.NotifyOperation(id, Delete);
         }
 
         public virtual void DidInsert(string id)
         {
-            NotifyOperation(id, Insert);
+            this.NotifyOperation(id, Insert);
         }
 
         public virtual void DidUpdate(string id)
         {
-            NotifyOperation(id, Update);
+            this.NotifyOperation(id, Update);
         }
 
         public void NotifyOperation(string id, string operation)
         {
-            var tableName = GetTableName();
+            var idType = typeof(TEntity)
+                .GetProperties()
+                .Where(p => p.Name.Equals("Id"))
+                .FirstOrDefault()
+                ?.PropertyType;
 
-            dynamic message = new ExpandoObject();
-            message.table = tableName;
-            message.id = id;
-            message.operation = operation;
 
-            var updateGroups = GetUpdateGroups(tableName, id);
-            var json = JsonConvert.SerializeObject(message);
-            SendStatusUpdateAsync(updateGroups, json).Wait();
+            dynamic idInEntityType; // Guid | int | string
```

Comment:
personally I would use object instead of dynamic here.  I dont think you are gaining anything by using a dynamic since DbSet().Find() takes an object[] as its params.  dynamic will allow you do things that cause runtime errors, like access properties that dont exist.